client/src/components/AuditAdmin/AuditAdminView.tsx

```diff
@@ -29,17 +30,34 @@ interface IParams {
 const AuditAdminView: React.FC = () => {
   const { electionId, view } = useParams<IParams>()
 
+  const queryClient = useQueryClient()
+  const history = useHistory()
+  const lastFetchedRounds = useRef<IRound[] | null>(null)
   const roundsQuery = useRounds(electionId, {
     refetchInterval: rounds =>
       rounds && isDrawingSample(rounds) ? 1000 : false,
+    onSuccess: rounds => {
+      // If we ever see the round status change from drawing to complete,
+      // redirect to the progress view and reload jurisdiction progress.
+      // This is a bit of a hacky way to do it, but there's not really a better
+      // way supported with react-query.
+      if (
+        lastFetchedRounds.current &&
+        !isDrawSampleComplete(lastFetchedRounds.current) &&
+        isDrawSampleComplete(rounds)
+      ) {
+        queryClient.invalidateQueries(jurisdictionsQueryKey(electionId))
+        history.push(`/election/${electionId}/progress`)
+      }
+      lastFetchedRounds.current = rounds
+    },
```

Comment:
This is a common problem I'm starting to see - how to trigger a frontend update when an "event" occurs on the server side. There's not really great support for this use case in the data fetching/state management libraries that I've seen. I'm curious to come up with a good approach that we can codify.